src/test/java/software/amazon/cloudformation/resource/IdentifierUtilsTest.java

```diff
@@ -48,4 +48,93 @@ public void generateResourceIdentifier_withShortLength_prefixTruncated() {
         // string that is left to fix the max length.
         assertThat(result).startsWith("my-re-");
     }
+
+    @Test
+    public void generateResourceIdentifier_withStackNameStackId() {
+        String result = IdentifierUtils.generateResourceIdentifier("my-stack-name", "my-resource", "123456", 18);
+        assertThat(result.length()).isLessThanOrEqualTo(18);
+
+        // to ensure randomness in the identity, the result will always be a random
+        // string PREFIXED by the size of
+        // string that is left to fix the max length.
+        assertThat(result).startsWith("my-m-");
+    }
+
+    @Test
+    public void generateResourceIdentifier_withStackName() {
+        String result = IdentifierUtils.generateResourceIdentifier("my-stack-name", "my-resource", "123456", 50);
+        assertThat(result.length()).isLessThanOrEqualTo(49);
+
+        // to ensure randomness in the identity, the result will always be a random
+        // string PREFIXED by the size of
+        // string that is left to fix the max length.
+        assertThat(result).startsWith("my-stack-name-my-resource-");
+    }
+
+    @Test
+    public void generateResourceIdentifier_withStackNameLessThanPreferredLen() {
+        String result = IdentifierUtils.generateResourceIdentifier("my-stack-name", "my-resource", "123456", 16);
+        assertThat(result.length()).isLessThanOrEqualTo(16);
+
+        // to ensure randomness in the identity, the result will always be a random
+        // string PREFIXED by the size of
+        // string that is left to fix the max length.
+        assertThat(result).startsWith("mym-");
+    }
+
+    @Test
+    public void generateResourceIdentifier_withStackNameBothFitMaxLen() {
+        String result = IdentifierUtils.generateResourceIdentifier(
+            "arn:aws:cloudformation:us-east-1:123456789012:stack/my-stack-name/084c0bd1-082b-11eb-afdc-0a2fadfa68a5",
+            "my-resource", "123456", 255);
+        assertThat(result.length()).isLessThanOrEqualTo(44);
+        assertThat(result).isEqualTo("my-stack-name-my-resource-hDoP0dahAFjd");
```

Comment:
I want to double check the random string generated, is it has both lower case and upper case? or just one of it?  It would be great to follow extract the same name pattern.